persistence-test/src/main/java/io/stargate/it/PersistenceTest.java

```diff
@@ -1666,7 +1666,7 @@ public void testWarnings() throws ExecutionException, InterruptedException {
     execute(conn, "USE " + ks.cqlName());
     execute(conn, "CREATE TABLE t (k int PRIMARY KEY)");
 
-    persistence().waitForSchemaAgreement();
+    conn.waitForSchemaAgreement();
```

Comment:
are we sure these 3 places are the only ones where we need to switch from `Persistence#waitForSchemaAgreement()` to `Connection#waitForSchemaAgreement`? 

Furthermore, is there a need now that `Persistence` also implements `SchemaAgreementChecker`? I guess yes because we also need to check if we are in schema agreement without actually having the connection, or? I wanted to double check, as if we can remove it we would be sure that all code has transitioned to the new way by a simple compile check..

Reply:
Updated the `DataStore`s which are the only users of the `waitForSchemaAgreement()` other than a test. This is used by the HTTP services (REST, GraphQL). CQL and gRPC use `Connection#isInSchemaAgreement()` directly.

Reply:
@mpenick one question regarding the direct usage of `Connection#isInSchemaAgreement()` in gRPC. Why it is used only in `QueryHandler`, but we don't need to use it in the `BatchHandler`? Can you elaborate more on this?